### PR TITLE
[Arch] split ArchAxis into separate modules

### PR DESCRIPTION
**Pull request by dirkolbrich**
Splitting the ArchAxis module into separate modules for ArchAxis, ArchAxisSystem and ArchGrid to aid in future dev and make the code base easier to understand.

See: https://forum.freecadweb.org/viewtopic.php?f=23&t=62575
See: https://forum.freecadweb.org/viewtopic.php?f=23&t=65063

---

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [x]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [x]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [x]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`

> Test WB fails in itself

- [x]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [X]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the [FreeCAD bug tracker issue number](https://freecadweb.org/wiki/tracker#GitHub_and_MantisBT) in case a particular commit solves or is related to an existing issue on the tracker. Ex: `Draft: fix typos - fixes #0004805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

File: src/Mod/Arch/ArchGrid.py [hunk 1]
```diff
@@ -0,0 +1,567 @@
...
+#***************************************************************************
+
+import FreeCAD,Draft,math,ArchCommands
+from FreeCAD import Vector
+if FreeCAD.GuiUp:
+    import FreeCADGui
+    from PySide import QtCore, QtGui
+    from DraftTools import translate
+    from pivy import coin
+    from PySide.QtCore import QT_TRANSLATE_NOOP
```
Comment (benj5378) [inline src/Mod/Arch/ArchGrid.py:29]: Unused modules

```suggestion
import FreeCAD,math
if FreeCAD.GuiUp:
    import FreeCADGui
    from PySide import QtCore, QtGui
    from DraftTools import translate
    from PySide.QtCore import QT_TRANSLATE_NOOP
```

File: src/Mod/Arch/ArchAxisSystem.py [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+#*                                                                         *
+#*   You should have received a copy of the GNU Library General Public     *
+#*   License along with this program; if not, write to the Free Software   *
+#*   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  *
+#*   USA                                                                   *
+#*                                                                         *
+#***************************************************************************
+
+import FreeCAD,Draft,math,ArchCommands
+from FreeCAD import Vector
```
Comment (benj5378) [inline src/Mod/Arch/ArchAxisSystem.py:23]: Unused modules

```suggestion
import FreeCAD
```

**Review by benj5378 — CHANGES_REQUESTED** [review]
Some unused modules related to the separation have to be removed.

There are a lot of other unused modules and other readability issues too in the files, that aren't related to the split up. I think it will be best to address those in a separate PR.

Good work!

**Comment by dirkolbrich** [comment]
Yes, this was my intention. First a split up, after that a clean up of the source. But you are right, removing any unused import resulting from the split should be in this PR. Will edit and submit again.

**Comment by dirkolbrich** [comment]
Another one would be removing the `six` module dependency, as this is used for Python 2 compatibility, which is with the v0.20 dev circle obsolete anyway?

https://github.com/FreeCAD/FreeCAD/blob/b3e52226fec48d58f4acca4f9699e52ed2e4c003/src/Mod/Arch/ArchAxis.py#L626-L627
https://github.com/FreeCAD/FreeCAD/blob/b3e52226fec48d58f4acca4f9699e52ed2e4c003/src/Mod/Arch/ArchAxis.py#L659-L660

**Comment by hyarion** [comment]
> as this is used for Python 2 compatibility, which is with the v0.20 dev circle obsolete anyway?

You are correct, Python 3.6+ is required for v0.20. So I agree that six could be removed.
It would be neat if that is done over the whole codebase in a separate PR. But fixing it here first doesn't hurt either.

**Comment by luzpaz** [comment]
In the past when we've gone to gung-ho on removing modules, things broke. I think removing modules, if it happens, should be in a separate PR so as to ensure that the code modifications in this PR are sound.

**Comment by dirkolbrich** [comment]
Agreed, removing the `six` dependency will be another PR.

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5400">pipeline status</a> for feature branch [PR_5400](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5400). Pipeline [454039958](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/454039958) was triggered at [a830e2c](https://github.com/FreeCAD/FreeCAD/pull/5400/commits/a830e2c329522164626cd5e9d5de33896f0668e0). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/Mod/Arch/ArchAxisSystem.py (hunk 2 above)
Comment (dirkolbrich) [reply src/Mod/Arch/ArchAxisSystem.py:23]: Not quite right. `Draft` is used within `_CommandAxisSystem`.

**Comment by dirkolbrich** [comment]
OK, I removed unused imports and reordered old inline imports to the files import block. Have a look and merge if ok.

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5400">pipeline status</a> for feature branch [PR_5400](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5400). Pipeline [454093401](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/454093401) was triggered at [7383a80](https://github.com/FreeCAD/FreeCAD/pull/5400/commits/7383a80b5396b1ecfa5d49ae334be3839f4fa965). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by yorikvanhavre** [comment]
Super work, thanks!
Regarding six/py2 deprecation, indeed we can do that as we go, but it's basically cleaning. It's always a good idea to do cleaning PRs separately, as they are easier to review and therefore much easier to merge

**Pull request merged**